sys/net/network_layer/ng_ipv6/netif/ng_ipv6_netif.c

```diff
@@ -440,14 +519,172 @@ static ipv6_addr_t *_match_prefix(kernel_pid_t pid, const ipv6_addr_t *addr,
     return NULL;
 }
 
-ipv6_addr_t *ng_ipv6_netif_match_prefix(kernel_pid_t pid, const ipv6_addr_t *prefix)
+/**
+ * @brief Determines the scope of the given address.
+ *
+ * @param[in] addr              The IPv6 address to check.
+ * @param[in] maybe_multicast   False if @p addr is definitely no multicast
+ *                              address, true otherwise.
+ *
+ * @return The scope of the address.
+ *
+ * @pre address is not loopback or unspecified.
+ * see http://tools.ietf.org/html/rfc6724#section-4
+ */
+static uint8_t _get_scope(const ipv6_addr_t *addr, const bool maybe_multicast)
+{
+    if (maybe_multicast && ipv6_addr_is_multicast(addr)) {
+        return (addr->u8[1] & 0x0f);
+    }
+    else if (ipv6_addr_is_link_local(addr)) {
+        return IPV6_ADDR_MCAST_SCP_LINK_LOCAL;
```

Comment:
Yes, I know. I had some dedicated defines before, but for easy usage they would have been a simple copy, so I decided against this duplication.